x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/helpers/generate_discovery.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+
+import {
+  AnalyticsServiceSetup,
+  AuthenticatedUser,
+  Logger,
+  SavedObjectsClientContract,
+} from '@kbn/core/server';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { AttackDiscoverGenerationConfig, Replacements } from '@kbn/elastic-assistant-common';
+import { PublicMethodsOf } from '@kbn/utility-types';
+import { ActionsClient } from '@kbn/actions-plugin/server';
+
+import { updateAttackDiscoveries } from './helpers';
+import { handleGraphError } from '../post/helpers/handle_graph_error';
+import { invokeAttackDiscoveryGraph } from '../post/helpers/invoke_attack_discovery_graph';
+import { AttackDiscoveryDataClient } from '../../../lib/attack_discovery/persistence';
+
+const ROUTE_HANDLER_TIMEOUT = 10 * 60 * 1000; // 10 * 60 seconds = 10 minutes
+const LANG_CHAIN_TIMEOUT = ROUTE_HANDLER_TIMEOUT - 10_000; // 9 minutes 50 seconds
+const CONNECTOR_TIMEOUT = LANG_CHAIN_TIMEOUT - 10_000; // 9 minutes 40 seconds
+
+export interface GenerateAttackDiscoveryParams {
+  actionsClient: PublicMethodsOf<ActionsClient>;
+  attackDiscoveryId: string;
+  authenticatedUser: AuthenticatedUser;
+  config: AttackDiscoverGenerationConfig;
+  dataClient: AttackDiscoveryDataClient;
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  savedObjectsClient: SavedObjectsClientContract;
+  telemetry: AnalyticsServiceSetup;
+}
+
+export const generateAttackDiscovery = async ({
+  actionsClient,
+  attackDiscoveryId,
```

Comment:
The name of this argument, `attackDiscoveryId` is an artifact of the current implementation. In the current implementation, this id is consistent (for a connector) across multiple runs.

In the near future, this value may be the same as `kibana.alert.rule.execution.uuid`.

Consider renaming this `attackDiscoveryId` to `executionUuid`, with the caveat that when this function is called by the legacy route (in the short term), the same UUID will be used across multiple runs.

Reply:
Would you want me to update the same name in `updateAttackDiscoveries` and `handleGraphError` or should we leave those unchanged for now?

Reply:
done in https://github.com/elastic/kibana/pull/216274/commits/198dabff5ecbfa14765daf988d230d5b304a3318